### PR TITLE
chore(ci): add sync branch workflow for storage-refactor

### PR DESCRIPTION
**Pull request by noahdietz**
Adds a new workflow that will send a PR week-daily at 4pm with changes from `main` to `storage-refactor`.

### CODE REVIEW

File: .github/workflows/sync_branch.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        run: echo ::set-output name=change_notes::$(git log storage-refactor..main --oneline --pretty=format:%s)
+      - name: Opening pull request
+        id: pull
+        # https://github.com/marketplace/actions/sync-branches
+        uses: tretuna/sync-branches@1.4.0
+        with:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          FROM_BRANCH: "main"
+          TO_BRANCH: "storage-refactor"
+          REVIEWERS: '["noahdietz"]'
```
Comment (tritone) [inline .github/workflows/sync_branch.yaml:29]: You can add me (or the dpe team) to reviewers as well, if you like.

**Review by tritone — APPROVED** [review]

File: .github/workflows/sync_branch.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  workflow_dispatch: {} # Allow manual triggering
+
+jobs:
+  sync-storage-refactor:
+    runs-on: ubuntu-latest
+    name: Syncing storage-refactor
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Node
```
Comment (codyoss) [inline .github/workflows/sync_branch.yaml:14]: Is this needed?

**Review by codyoss — APPROVED** [review]
LGTM, just one question

File: .github/workflows/sync_branch.yaml (hunk 2 above)
Comment (noahdietz) [reply .github/workflows/sync_branch.yaml:14]: I'm not sure, it is from the README of the Action: https://github.com/marketplace/actions/sync-branches. So I'm guessing yes?

**Pull request merged**